lib/styles.js

```diff
@@ -89,25 +89,34 @@ module.exports = function (expect) {
             .jsString("'");
     });
 
-    expect.addStyle('key', function (content) {
+    expect.addStyle('key', function (key, isArrayLike) {
         var isSymbol;
         /*jshint ignore:start*/
-        isSymbol = typeof content === 'symbol';
+        isSymbol = typeof key === 'symbol';
         /*jshint ignore:end*/
         if (isSymbol) {
-            this.text('[').sp().appendInspected(content).sp().text(']');
+            this.text('[').sp().appendInspected(key).sp().text(']').text(':');
         } else {
-            content = String(content);
-            if (/^[a-z\$\_][a-z0-9\$\_]*$/i.test(content)) {
-                this.text(content, 'jsKey');
-            } else if (/^(?:0|[1-9][0-9]*)$/.test(content)) {
-                this.jsNumber(content);
+            key = String(key);
+            if (/^[a-z\$\_][a-z0-9\$\_]*$/i.test(key)) {
+                this.text(key, 'jsKey').text(':');
+            } else if (/^(?:0|[1-9][0-9]*)$/.test(key)) {
+                if (!isArrayLike) {
+                    this.jsNumber(key).text(':');
+                }
             } else {
-                this.singleQuotedString(content);
+                this.singleQuotedString(key).text(':');
             }
         }
     });
 
+    expect.addStyle('nonLeadingSpace', function (num) {
+        var lastLine = this.output[this.output.length - 1];
+        if (lastLine.length > 0 && !/\s$/.test(lastLine[lastLine.length - 1].args.content)) {
+            this.sp(num);
+        }
+    });
```

Comment:
Yeah, good shout. I meant to find a better solution for that, but never got back to it.

The problem it's currently "solving" is that the `key` style might or might not output something when in `array` mode, and when it's not outputting something, the space "after" it should be suppressed. I first tried building the space into the style, and that worked fine, except it caused trailing whitespace in the tests where the value is multi-line or wide so a newline has to be inserted.

Either I could try taking a step back so the `key` style does fewer things (effectively adding a check for whether the key is numeric to every place where the `key` style is used by the `array-like` type), or I could try to make a `keyValue` style that renders both the key and the value with the correct spacing in between. I recall stumbling upon a few places where a newline didn't get inserted before a multi-line or wide value, so that approach might help us get that behavior applied everywhere.

What do you think?